chainer/functions/connection/bilinear.py

```diff
@@ -51,6 +51,11 @@ def forward(self, inputs):
         e2 = array.as_mat(inputs[1])
         W = inputs[2]
 
+        if(not type_check.same_types(*inputs)):
+            raise ValueError('numpy and cupy must not be used together\n'
+                             'type(W): %s, type(e1): %s, type(e2): %s'
+                             % (type(W), type(e1), type(e2)))
```

Comment:
Formatting with `%` operator is discouraged. Could you use `str.format()` instead?